src/main/java/io/strimzi/utils/LogicalKafkaVersionEntity.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.utils;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.testcontainers.shaded.com.fasterxml.jackson.databind.JsonNode;
+import org.testcontainers.shaded.com.fasterxml.jackson.databind.ObjectMapper;
+import org.testcontainers.shaded.org.apache.commons.io.IOUtils;
+
+import java.io.IOException;
+import java.io.StringWriter;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ *  A logical Kafka version
+ */
+public class LogicalKafkaVersionEntity {
```

Comment:
"Entity" doesn't really explain what's going on here. The inner class represents a concrete "Kafka version" and the outer class is more like a service that allows you to discover a concrete version that satisfies some more abstract criteria. So I would rename `LogicalKafkaVersionEntity` → `KafkaVersionService` and `LogicalKafkaVersion` → `KafkaVersion`. I think that naming would make it a bit clearer what's going on.